crates/runestick/src/compile_meta.rs

```diff
@@ -194,6 +197,9 @@ pub enum CompileMetaKind {
     ConstFn {
         /// Opaque identifier for the constant function.
         id: Id,
+
+        /// Whether this function has a test annotation
+        is_test: bool,
```

Comment:
@udoprog I think you had opinions on this. See https://github.com/rust-lang/rust/issues/51999 and related threads for rust background on static contexts and asserts. Seems like a reduced feature set is supported, which allows implementing tests as const fn. Not sure whether it is required to interact with the test system - I guess by definition any assert in const code blocks the build, which precludes testing it. It'd just become weird if you *want* to test the code and you end failing tests during compilation...